salt/modules/saltutil.py

```diff
@@ -1444,6 +1445,14 @@ def _exec(client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, **kwargs):
             'ret': ret, 'cli': True, 'kwarg': kwarg, 'sub': kwargs['subset'],
         }
         del kwargs['subset']
+    elif kwargs.get('asynchronous'):
```

Comment:
We should change the function signature to:

`def _exec(client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, batch=False, subset=False, asynchronous=False, **kwargs):`

That way we don't need to remove the flags from kwargs manually - any excess options will be passed to the command.

Reply:
Doing that would also require removing a few lines from `saltmod.py:function` which determines if batch, subset and asynchronous are in kwargs.